static/js/admin/src/AddMenu.js

```diff
@@ -9,19 +9,56 @@ define('AddMenu', [ 'jquery', 'jquery-ui', 'underscore', 'CreateContainerForm',
 	
 	AddMenu.prototype.getMenuItems = function() {
 		var items = {};
+
 		if ($.inArray('addRemoveContents', this.container.permissions) != -1) {
-			items["addContainer"] = {name : "Add Container"};
-			items["ingestPackage"] = {name : "Add Ingest Package"};
-			items["ingestSource"] = {name : "Add from File Server"};
-			items["simpleObject"] = {name : "Add Simple Object"};
-			
+			var self = this;
+
+			items["addContainer"] = {
```

Comment:
This button is recreated after a page navigation, so it would also be okay to have an if statement that determines if the options gets added rather than a callback function, so you could cut down on the amount of code here.  

Also, the isVisible method is a bit of a misnomer since it is really "isAggregate", which is the what's actually being evaluated for all of the "add" options.

Additionally, I can't tell if this is being handled another way since I can't deploy the application at the moment, but none of these options should be visible if the self.container.type == "File" (or simple or whatever its called)